### PR TITLE
Issue 2331: Enhanced test case for resolving product deps

### PR DESCRIPTION
**Pull request by BenShelbourne**

### CODE REVIEW

**Comment by laeubi** [comment]
@BenShelbourne thanks for enhancing the testcase, I'll try to look into this the next days.

**Comment by laeubi** [comment]
@BenShelbourne can you check the ECA did you used the same mail address as the one you signed the ECA?

**Comment by laeubi** [comment]
@BenShelbourne the testcase now passes on the Jenkins CI so I think it would be good to be merged but we need a valid ECA check before

**Comment by BenShelbourne** [comment]
@laeubi sorry for the delay on this. I'm still awaiting approval from our legal team to check I am okay to sign it. There are people already with approval so I can't imagine it will be much longer until I get it! Thanks for the fix and sorry about the delay!!

**Comment by BenShelbourne** [comment]
I've tested the latest changes and can confirm everything builds perfectly as expected so thank you for addressing this issue. I hopefully will sign the ECA soon and we can get this merged!

**Comment by BenShelbourne** [comment]
I've now signed the ECA. Sorry about the delay!

**Comment by laeubi** [comment]
@BenShelbourne mayn thanks the ECA works now but it seems there is a problem with the test now on macos/windows:

```
Caused by: org.apache.maven.plugin.MojoExecutionException: Detected Luna or older launcher feature org.eclipse.equinox.executable version 3.3.101.R34x_v20081125-7H-ELfE8hXnkE15Wh9Tnyu.
 Native product launchers for MacOSX can only be built against Eclipse Mars or newer.
To fix this, you can either build against Eclipse Mars or newer (recommended) or go back to Tycho <= 0.22.0
    at org.eclipse.tycho.plugins.p2.publisher.PublishProductMojo.checkMacOSLauncherCompatibility (PublishProductMojo.java:156)
    at org.eclipse.tycho.plugins.p2.publisher.PublishProductMojo.getExpandedLauncherBinaries (PublishProductMojo.java:121)
    at org.eclipse.tycho.plugins.p2.publisher.PublishProductMojo.publishContent (PublishProductMojo.java:102)
    at org.eclipse.tycho.plugins.p2.publisher.AbstractPublishMojo.execute (AbstractPublishMojo.java:39)
```

I think the easiest for this purpose would be to not include the native launcher in the build product.

File: tycho-its/projects/issue2331/product/issue2331.product [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?pde version="3.5"?>
+
+<product uid="issue2331" version="0.0.1.qualifier" useFeatures="false" includeLaunchers="true">
```
Comment (laeubi) [inline tycho-its/projects/issue2331/product/issue2331.product:4]:
```suggestion
<product uid="issue2331" version="0.0.1.qualifier" useFeatures="false" includeLaunchers="false">
```

**Pull request merged**